controllers/dockermachine_controller.go

```diff
@@ -115,6 +118,38 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 		return ctrl.Result{}, err
 	}
 
+
+	// If the DockerMachine doesn't have finalizer, add it.
+	if !util.Contains(dockerMachine.Finalizers, capiv1alpha2.MachineFinalizer) {
+		dockerMachine.Finalizers = append(dockerMachine.Finalizers, capiv1alpha2.MachineFinalizer)
+		if err := r.Client.Patch(ctx, dockerMachine, patch) ; err!= nil {
+
+			return ctrl.Result{}, errors.Wrapf(err, "failed to add finalizer to DockerMachine %q in namespace %q",
+				dockerMachine.Name, dockerMachine.Namespace)
+		}
+
+		// Since adding the finalizer updates the object return to avoid later update issues
```

Comment:
We probably don't need to do this, it was needed in the controller/actuator model we had for v1alpha1, but shouldn't be needed anymore since this controller is the only one updating the DockerMachine resources.